src/ukify/ukify.py

```diff
@@ -1053,56 +1041,22 @@ def pack_strings(strings: set[str], base: int) -> tuple[bytes, dict[str, int]]:
 
 
 def parse_hwid_dir(path: Path) -> bytes:
-    hwid_files = path.rglob('*.txt')
+    hwid_files = path.rglob('*.json')
 
     strings: set[str] = set()
-    devices: collections.defaultdict[tuple[str, str], list[EFI_GUID]] = collections.defaultdict(list)
-
-    uuid_regexp = re.compile(
-        r'\{[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}\}', re.I
-    )
+    devices: collections.defaultdict[tuple[str, str], list[uuid.UUID]] = collections.defaultdict(list)
 
     for hwid_file in hwid_files:
-        content = hwid_file.open().readlines()
-
-        data: dict[str, str] = {
-            'Manufacturer': '',
-            'Family': '',
-            'Compatible': '',
-        }
-        uuids: list[EFI_GUID] = []
+        data = json.loads(hwid_file.read_text(encoding='UTF-8'))
 
-        for line in content:
-            for k in data:
-                if line.startswith(k):
-                    data[k] = line.split(':')[1].strip()
-                    break
-            else:
-                uuid = uuid_regexp.match(line)
-                if uuid is not None:
-                    d1, d2, d3, d4, d5 = uuid.group(0)[1:-1].split('-')
-
-                    data1 = int(d1, 16)
-                    data2 = int(d2, 16)
-                    data3 = int(d3, 16)
-                    data4 = cast(
-                        tuple[int, int, int, int, int, int, int, int],
-                        tuple(hex_pairs_list(d4) + hex_pairs_list(d5)),
-                    )
-
-                    uuids.append((data1, data2, data3, data4))
-
-        for k, v in data.items():
-            if not v:
+        for k in ['name', 'compatible', 'hwids']:
+            if k not in data:
                 raise ValueError(f'hwid description file "{hwid_file}" does not contain "{k}"')
 
-        name = data['Manufacturer'] + ' ' + data['Family']
-        compatible = data['Compatible']
-
-        strings |= set([name, compatible])
+        strings |= set([data['name'], data['compatible']])
```

Comment:
This can use a set literal.
```suggestion
        strings |= {data['name'], data['compatible']}
```

Reply:
Also, is it necessary to have the separate `strings` set? Looking at `pack_strings`, there everything gets jumbled up through the sorting, i.e. the values of `name` and `compatible` (though I have no idea what they look like) could be spaced arbitrarily far apart and in `pack_device` one than gets the offsets into the output of `pack_strings` to recover them. Instead, one could use that `devices.keys()` is already a set (well, a dict view, but that behaves like a set) of string pairs and use that instead. Sorting that would sort the string pairs and the resulting offsets from `pack_strings` would then always be close together.

Reply:
Example JSON files are available here: https://github.com/anonymix007/systemd-stub/tree/master/json

The point of the separate `strings` set is to deduplicate these strings. Consider 2 SKUs: both would have the same `name` (defined by the manufacturer) and `compatible` (defined by the kernel). We can also end up in a situation when only one of those could be duplicate. For example, 'Laptop X' will have 2 different SKUs which require different `compatibles` due to slight hardware differences such as display, or maybe the hardware is the same, but the model is called differently in different regions.

Reply:
I see, that makes sense. I have no idea whether the win in space saving is worth the loss of locality, so refer to your judgement there. Still think the set literal is easier on the eyes though. :)

Reply:
The examples are quite helpful, maybe a smaller one of those would be helpful in the man page.

Reply:
I don't think locality matters here at all. In fact, only `compatible` is used currently used by the stub.